app/assets/javascripts/dashboard/views/dash/explorative_annotations_view.js

```diff
@@ -241,21 +258,73 @@ export default class ExplorativeAnnotationsView extends React.PureComponent {
         Toast.message(data.messages);
 
         this.setState({
-          archivedTracings: this.state.unarchivedTracings.concat(this.state.archivedTracings),
-          unarchivedTracings: [],
+          archivedTracings: this.state.archivedTracings.concat(selectedAnnotations),
+          unarchivedTracings: _.without(this.state.unarchivedTracings, ...selectedAnnotations),
         });
       },
     });
   };
 
+  addTagToSearch = (tag: string): void => {
+    if (!this.state.tags.includes(tag)) {
+      const newTags = update(this.state.tags, { $push: [tag] });
+      this.setState({ tags: newTags });
+      localStorage.setItem("lastDashboardSearchTags", JSON.stringify(newTags));
+    }
+  };
+
+  removeTagFromSearch = (tag: string): void => {
+    const newTags = this.state.tags.filter(t => t !== tag);
+    this.setState({ tags: newTags });
+    localStorage.setItem("lastDashboardSearchTags", JSON.stringify(newTags));
+  };
+
+  addTagToAnnotation = (annotation: APIAnnotationType, newTag: string): void => {
+    const newTracings = this.state.unarchivedTracings.map(t => {
```

Comment:
You have to check here whether unarchived or archived tracings are currnetly shown. At the moment, archived tracings cannot receive a new tag.